File: example/lua/cfg/equip/jewelrytype.lua
local cfg = require "cfg._cfgs"

---@class cfg.equip.jewelrytype
---@field typeName string , 程序用名字
---@field get fun(TypeName:string):cfg.equip.jewelrytype
---@field Jade cfg.equip.jewelrytype
---@field Bracelet cfg.equip.jewelrytype
---@field Magic cfg.equip.jewelrytype
---@field Bottle cfg.equip.jewelrytype
---@field all table<any,cfg.equip.jewelrytype>

local this = cfg.equip.jewelrytype

local mk = cfg._mk.table(this, { { 'all', 'get', 1 }, }, 1, nil, 
    'typeName' -- string, 程序用名字
    )

mk("Jade")
mk("Bracelet")
mk("Magic")
mk("Bottle")

return this

File: example/java/config/equip/Jewelrytype.java
package config.equip;

public enum Jewelrytype {
    JADE("Jade"),
    BRACELET("Bracelet"),
    MAGIC("Magic"),
    BOTTLE("Bottle");

    private String value;

    Jewelrytype(String value) {
        this.value = value;
    }

    private static java.util.Map<String, Jewelrytype> map = new java.util.HashMap<>();

    static {
        for(Jewelrytype e : Jewelrytype.values()) {
            map.put(e.value, e);
        }
    }

    public static Jewelrytype get(String value) {
        return map.get(value);
    }

    /**
     * 程序用名字
     */
    public String getTypeName() {
        return value;
    }

}
